lymph/node.py

```diff
@@ -9,23 +9,25 @@ class Node(object):
     system. This serves as one part of a lightweight network setup (the other
     one is the :class:`Edge` class).
     """
-    def __init__(self, name: str, state: int = 0, typ: str = "lnl"):
+    def __init__(self, name: str, state: int = 0, typ: str = "lnl", allowed_states: int = 3, growth_parameter: float = 0):
         """
         Args:
             name: Name of the node.
-            state: Current state this LNL is in. Can be in {0, 1}.
+            state: Current state this LNL is in. Can be in any allowed state e.g. for 3: {0, 1, 2}.
             typ: Can be either ``"lnl"``, ``"tumor"``.
         """
+        self.allowed_states = int(allowed_states)
         if type(name) is not str:
             raise TypeError("Name of node must be a string")
-        if int(state) not in [0,1]:
-            raise ValueError("State must be castable to 0 or 1")
+        if int(state) not in list(range(allowed_states)):
+            raise ValueError("State must be castable to the allowed states")
         if typ not in ["lnl", "tumor"]:
             raise ValueError("Typ of node must be either `lnl` or `tumor`")
 
         self.name = name
         self.typ = typ
         self.state = int(state)
+        self.growth_parameter = float(growth_parameter)

```

Comment:
Following the reasoning in #44, this should only appear in the `Edge` class.

Reply:
Only if we produce a self edge. The growth parameter is not assigned to an edge. In my opinion the growth parameter belongs to a node or to an edge that connects the node with itself. But we discussed that a self connecting edge would bring new problems, didn't we?

Reply:
We did, but I think it'd actually be better if we create edges that have the same node as start and end. For example, we could add a `@property` to the `Edge` class like this:

```python
@property
def is_growth(self) -> bool:
    """Check if this edge represents a node's grwoth."""
    return self.start == self.end
```

And then we use something like

```python
if incomming.is_growth:
    ...
```

in the `Node` class for the case that we want to compute a transition probability from 1 to 2.

The nice thing about this is on the one hand that it makes sense: There _is_ an edge, and even in your slides you drew one. So, we can just tell potential users: "Every arc in this diagram corresponds to one `Edge` instance."

On the other hand, it gives us the flexibility to assign different growth parameters to different nodes. Or even entirely different behaviours. Looking at the project right now, this does not seem to make a whole lot of sense but who knows, we might want to model breast or prostate cancer with this one day and dynamics might be different there.

Also, in the end, this parametrization just happens to be interpretable using the neck's anatomy. But it's not a simulation and if parameters don't make intuitive sense, but describe the data better, then dropping the mechanistic view of it is probably the right thing to do. We kind of did that already with the arc from LNL II to LNL I.